group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3835,24 +4134,35 @@ public void replay(
     public void onNewMetadataImage(MetadataImage newImage, MetadataDelta delta) {
         metadataImage = newImage;
 
+        // Initialize the last offset if it was not yet.
+        if (lastMetadataImageWithNewTopics == -1L) {
+            lastMetadataImageWithNewTopics = metadataImage.provenance().lastContainedOffset();
+        }
+
+        TopicsDelta topicsDelta = delta.topicsDelta();
+        if (topicsDelta == null) return;
+
+        // Updated the last offset of the image with newly created topics.
+        if (!topicsDelta.createdTopicIds().isEmpty()) {
```

Comment:
i feel like i'm missing something - don't we need to update the last offset for deleted topics as well? i thought the `deletedTopicIds()` iteration below implicitly triggers the regex refresh but that only refreshes subscribed topic names right?

Reply:
That's right. I have opted for non triggering a refresh of the regexes on deleted topics but to only do it on created topics. Deleted topics are removed from the subscription metadata and the assignment via the `requestMetadataRefresh` path. However, they would stay in the resolved regexes until they are refreshed. They don't really hurt because they don't exist anymore. Hence, I feel that it is an acceptable tradeoff to make in order to reduce the rate of refreshes. What do you think?

Reply:
Thanks, that makes sense. Could we briefly mention it in the comments here?